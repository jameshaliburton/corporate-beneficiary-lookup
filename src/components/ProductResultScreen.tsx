import React, { useState } from 'react';

export interface ProductHeader {
  name: string;
  brand: string;
  country: string;
  flag: string;
  barcode?: string;
}

export interface OwnershipStep {
  name: string;
  country: string;
  flag: string;
  type: string;
  ultimate?: boolean;
}

export interface ProcessTraceRow {
  step: string;
  status: string;
  reasoning: string;
  time: string;
}

export interface ProductResultScreenProps {
  product?: ProductHeader;
  ownershipTrail?: OwnershipStep[];
  confidence?: 'High' | 'Medium' | 'Low';
  confidenceScore?: number;
  sources?: string[];
  processTrace?: ProcessTraceRow[];
  error?: string | null;
  fallback?: string | null;
}

// Mock data for demonstration
const mockResult: ProductResultScreenProps = {
  product: {
    name: 'Kit Kat Matcha Green Tea',
    brand: 'Nestlé',
    country: 'Japan',
    flag: '🇯🇵',
    barcode: '4902201176291',
  },
  ownershipTrail: [
    { name: 'Nestlé Japan', country: 'Japan', flag: '🇯🇵', type: 'Private' },
    { name: 'Nestlé S.A.', country: 'Switzerland', flag: '🇨🇭', type: 'Public', ultimate: true },
  ],
  confidence: 'High',
  confidenceScore: 92,
  sources: ['OpenCorporates', 'Nestlé Annual Report'],
  processTrace: [
    { step: 'Cache Check', status: 'MISS', reasoning: 'No cached result found', time: '148ms' },
    { step: 'Static Mapping', status: 'MISS', reasoning: 'No static match', time: '155ms' },
    { step: 'Query Builder', status: '✅', reasoning: 'Generated search queries', time: '2.1s' },
    { step: 'Web Research', status: '✅', reasoning: 'Scraped and parsed web info', time: '41s' },
    { step: 'Ownership Agent', status: '✅', reasoning: 'LLM-based ownership analysis', time: '43s' },
    { step: 'Validation', status: '✅', reasoning: 'Result validated + deduplicated', time: '43s' },
    { step: 'DB Save', status: '✅', reasoning: 'Saved to DB', time: '~' },
  ],
  error: null,
  fallback: null,
};

const confidenceColors = {
  High: 'bg-blue-100 text-blue-800',
  Medium: 'bg-yellow-100 text-yellow-800',
  Low: 'bg-red-100 text-red-800',
};

const typeColors = {
  Public: 'bg-blue-100 text-blue-800',
  Private: 'bg-gray-100 text-gray-800',
  'State-owned': 'bg-green-100 text-green-800',
};

const ProductResultScreen: React.FC<ProductResultScreenProps> = (props) => {
  const [showTrace, setShowTrace] = useState(false);
  const [showManualEntry, setShowManualEntry] = useState(false);

  // Use props if provided, otherwise fallback to mock data
  const {
    product = mockResult.product,
    ownershipTrail = mockResult.ownershipTrail,
    confidence = mockResult.confidence,
    confidenceScore = mockResult.confidenceScore,
    sources = mockResult.sources,
    processTrace = mockResult.processTrace,
    error = mockResult.error,
    fallback = mockResult.fallback,
  } = props;

  return (
    <div className="flex flex-col min-h-screen bg-gray-50 pb-24">
      {/* 🛍️ Product Header */}
      <section className="bg-white rounded-xl shadow p-6 mb-4 mt-4 mx-2">
        <div className="flex flex-col items-center text-center">
          <span className="text-3xl font-bold mb-1">{product.name}</span>
          <span className="text-lg text-gray-600 mb-1">{product.brand}</span>
          <span className="text-base flex items-center gap-1 mb-1">{product.flag} {product.country}</span>
          {product.barcode && <span className="text-xs text-gray-400">Barcode: {product.barcode}</span>}
        </div>
      </section>

      {/* 🧬 Ownership Trail */}
      <section className="bg-white rounded-xl shadow p-4 mb-4 mx-2">
        <div className="mb-2 font-semibold text-lg flex items-center gap-2">🧬 Ownership</div>
        <div className="flex flex-wrap gap-2 justify-center">
          {ownershipTrail.map((step, idx) => (
            <div key={idx} className={`flex items-center gap-1 px-3 py-2 rounded-full shadow-sm border ${typeColors[step.type] || 'bg-gray-100 text-gray-800'}`}
              style={{ fontWeight: step.ultimate ? 'bold' : 'normal', borderWidth: step.ultimate ? 2 : 1 }}>
              <span>{step.name}</span>
              <span>{step.flag}</span>
              <span className="ml-1 px-2 py-0.5 rounded text-xs font-medium bg-white border border-gray-200">{step.type}</span>
              {step.ultimate && <span className="ml-1">🎯 <span className="text-xs bg-green-100 text-green-800 px-2 py-0.5 rounded-full ml-1">Ultimate Owner</span></span>}
              {idx < ownershipTrail.length - 1 && <span className="mx-2 text-gray-400">→</span>}
            </div>
          ))}
        </div>
      </section>

      {/* 📊 Confidence + Attribution */}
      <section className="bg-white rounded-xl shadow p-4 mb-4 mx-2">
        <div className="mb-2 font-semibold text-lg flex items-center gap-2">📊 Confidence</div>
        <div className="flex items-center gap-2">
          <span className={`px-3 py-1 rounded-full font-semibold text-sm ${confidenceColors[confidence]}`}>{confidence} ({confidenceScore}%)</span>
          <button className="ml-2 text-xs underline text-blue-600" title="Confidence is based on data quality, source reliability, and agent agreement.">What affects this?</button>
        </div>
        <div className="mt-2 flex gap-2">
          <button className="px-3 py-1 rounded bg-green-100 text-green-800 font-medium text-xs">🙋 Confirmed</button>
          <button className="px-3 py-1 rounded bg-red-100 text-red-800 font-medium text-xs">❗ Flag Incorrect</button>
        </div>
      </section>

      {/* 🧪 Process Trace */}
      <section className="bg-white rounded-xl shadow p-4 mb-4 mx-2">
        <div className="flex items-center justify-between mb-2">
          <span className="font-semibold text-lg flex items-center gap-2">🧪 Trace</span>
          <button className="text-xs underline text-blue-600" onClick={() => setShowTrace(!showTrace)}>{showTrace ? 'Hide' : 'Show'} Steps</button>
        </div>
        {showTrace && (
          <div className="overflow-x-auto">
            <table className="min-w-full text-sm">
              <thead>
                <tr className="bg-gray-100">
                  <th className="px-2 py-1 text-left">Step</th>
                  <th className="px-2 py-1 text-left">Status</th>
                  <th className="px-2 py-1 text-left">Reasoning</th>
                  <th className="px-2 py-1 text-left">Time</th>
                </tr>
              </thead>
              <tbody>
                {processTrace.map((row, idx) => (
                  <tr key={idx} className="border-b last:border-b-0">
                    <td className="px-2 py-1 whitespace-nowrap">{row.step}</td>
                    <td className="px-2 py-1">{row.status}</td>
                    <td className="px-2 py-1">{row.reasoning}</td>
                    <td className="px-2 py-1">{row.time}</td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        )}
      </section>

      {/* 🧯 Error States + Fallback */}
      {error && (
        <section className="bg-white rounded-xl shadow p-4 mb-4 mx-2">
          <div className="font-semibold text-lg flex items-center gap-2 text-red-700">❌ Error</div>
          <div className="text-red-700 mt-2">{error}</div>
          <button className="mt-4 px-4 py-2 rounded bg-blue-100 text-blue-800 font-medium" onClick={() => window.location.reload()}>🔄 Retry</button>
        </section>
      )}
      {fallback && (
        <section className="bg-white rounded-xl shadow p-4 mb-4 mx-2">
          <div className="font-semibold text-lg flex items-center gap-2">🧯 Fallback</div>
          <div className="text-gray-700 mt-2">{fallback}</div>
        </section>
      )}

      {/* 📝 Manual Entry Form */}
      {showManualEntry && (
        <section className="bg-white rounded-xl shadow p-4 mb-4 mx-2">
          <div className="font-semibold text-lg flex items-center gap-2">📝 Manual Entry</div>
          <form className="mt-2 flex flex-col gap-3">
            <input className="border rounded px-3 py-2" placeholder="Product Name" />
            <input className="border rounded px-3 py-2" placeholder="Brand Name" />
            <input className="border rounded px-3 py-2" placeholder="Company Name" />
            <input className="border rounded px-3 py-2" placeholder="Country (optional)" />
            <button className="mt-2 px-4 py-2 rounded bg-blue-600 text-white font-semibold">Re-run Search</button>
            <label className="flex items-center gap-2 text-xs mt-2">
              <input type="checkbox" /> Contribute data
            </label>
          </form>
        </section>
      )}

      {/* Sticky Action Bar */}
      <div className="fixed bottom-0 left-0 w-full bg-white border-t shadow-lg flex justify-around py-3 z-50">
        <button className="flex-1 mx-2 px-3 py-2 rounded bg-blue-100 text-blue-800 font-medium">🔄 Retry</button>
        <button className="flex-1 mx-2 px-3 py-2 rounded bg-green-100 text-green-800 font-medium">🙋 Confirm</button>
        <button className="flex-1 mx-2 px-3 py-2 rounded bg-red-100 text-red-800 font-medium">❗ Flag</button>
        <button className="flex-1 mx-2 px-3 py-2 rounded bg-gray-100 text-gray-800 font-medium" onClick={() => setShowManualEntry(true)}>📝 Enter Manually</button>
      </div>
    </div>
  );
};

export default ProductResultScreen; 