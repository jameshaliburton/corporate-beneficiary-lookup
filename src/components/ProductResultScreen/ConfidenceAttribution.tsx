import React from 'react';

export interface ConfidenceAttributionProps {
  confidence: 'High' | 'Medium' | 'Low';
  score?: number;
  onConfirm?: () => void;
  onFlag?: () => void;
}

const confidenceColors: Record<string, { bg: string; color: string }> = {
  High: { bg: '#e0f2fe', color: '#2563eb' }, // blue
  Medium: { bg: '#fef9c3', color: '#ca8a04' }, // yellow
  Low: { bg: '#fee2e2', color: '#dc2626' }, // red
};

const ConfidenceAttribution: React.FC<ConfidenceAttributionProps> = ({ confidence, score, onConfirm, onFlag }) => {
  const { bg, color } = confidenceColors[confidence] || confidenceColors.High;
  return (
    <div
      style={{
        background: '#fff',
        borderRadius: 16,
        boxShadow: '0 2px 8px rgba(0,0,0,0.04)',
        padding: '1.25rem 1rem',
        marginBottom: '1.5rem',
        maxWidth: 420,
        marginLeft: 'auto',
        marginRight: 'auto',
      }}
    >
      <div style={{ fontWeight: 600, fontSize: 18, marginBottom: 10, display: 'flex', alignItems: 'center', gap: 8 }}>
        <span role="img" aria-label="confidence">üìä</span> Confidence
      </div>
      <div style={{ display: 'flex', alignItems: 'center', gap: 12, marginBottom: 8 }}>
        <span
          style={{
            background: bg,
            color,
            fontWeight: 600,
            fontSize: 15,
            borderRadius: 8,
            padding: '4px 14px',
            display: 'inline-block',
          }}
        >
          {confidence} {score !== undefined ? `(${score}%)` : ''}
        </span>
        <a
          href="#"
          style={{ fontSize: 13, color: '#2563eb', textDecoration: 'underline', cursor: 'pointer' }}
          title="Learn what affects confidence"
        >
          What affects this?
        </a>
      </div>
      <div style={{ display: 'flex', gap: 10, marginTop: 8 }}>
        <button
          onClick={onConfirm}
          style={{
            background: '#dcfce7',
            color: '#166534',
            fontWeight: 600,
            border: 'none',
            borderRadius: 8,
            padding: '7px 16px',
            fontSize: 15,
            cursor: 'pointer',
            flex: 1,
          }}
        >
          üôã Confirmed
        </button>
        <button
          onClick={onFlag}
          style={{
            background: '#fee2e2',
            color: '#b91c1c',
            fontWeight: 600,
            border: 'none',
            borderRadius: 8,
            padding: '7px 16px',
            fontSize: 15,
            cursor: 'pointer',
            flex: 1,
          }}
        >
          ‚ùó Flag Incorrect
        </button>
      </div>
    </div>
  );
};

export default ConfidenceAttribution; 