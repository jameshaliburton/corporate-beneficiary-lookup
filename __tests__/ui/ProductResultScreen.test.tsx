import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import ProductResultScreen from '@/components/ProductResultScreen';
import {
  mockSuccessResult,
  mockAmbiguousResult,
  mockBadJSONResult,
  mockTimeoutResult,
  mockUnknownResult,
  mockFollowUpContextResult
} from '../../__mocks__/ownershipResults';
import {
  mockTraceManualEntry,
  mockTraceRetry,
  mockTraceVerified,
  mockTraceUnverified,
  mockTraceUnknown,
  mockTraceAmbiguous
} from '../../__mocks__/traceMocks';

describe('ProductResultScreen', () => {
  const baseProps = {
    product: {
      name: 'Test Product',
      brand: 'Test Brand',
      country: 'United States',
      flag: '🇺🇸',
      barcode: '123456789'
    },
    ownershipTrail: [
      {
        name: 'Test Brand',
        country: 'United States',
        flag: '🇺🇸',
        type: 'Brand'
      },
      {
        name: 'Parent Company Inc',
        country: 'United States',
        flag: '🇺🇸',
        type: 'Parent Company'
      }
    ],
    confidence: 'High' as const,
    confidenceScore: 92,
    sources: ['https://example.com', 'https://parent.com'],
    processTrace: [
      { step: 'Web Search', status: '✅', reasoning: 'Found 3 sources', time: '2.1s' },
      { step: 'Analysis', status: '✅', reasoning: 'Extracted ownership chain', time: '43s' }
    ],
    error: null,
    fallback: null
  };

  it('renders ownership chain and confidence pill', () => {
    render(<ProductResultScreen {...baseProps} />);
    expect(screen.getAllByText('Test Brand')).toHaveLength(2);
    expect(screen.getByText('Parent Company Inc')).toBeInTheDocument();
    expect(screen.getByText('Brand')).toBeInTheDocument();
    expect(screen.getByText('Parent Company')).toBeInTheDocument();
    expect(screen.getByText(/High \(92%\)/)).toBeInTheDocument();
  });

  it('shows unknown owner fallback state', () => {
    const unknownProps = {
      ...baseProps,
      ownershipTrail: [],
      confidence: 'Low' as const,
      confidenceScore: 0,
      processTrace: [
        { step: 'Web Search', status: '❌', reasoning: 'No sources found', time: '1.2s' }
      ]
    };
    render(<ProductResultScreen {...unknownProps} />);
    expect(screen.getByText('🧬 Ownership')).toBeInTheDocument();
    expect(screen.getByText(/Low \(0%\)/)).toBeInTheDocument();
  });

  it('displays trace steps in correct order with statuses', () => {
    render(<ProductResultScreen {...baseProps} />);
    const showButton = screen.getByText(/Show Steps/i);
    fireEvent.click(showButton);
    expect(screen.getByText('Web Search')).toBeInTheDocument();
    expect(screen.getByText('Analysis')).toBeInTheDocument();
    expect(screen.getByText('Found 3 sources')).toBeInTheDocument();
    expect(screen.getByText('Extracted ownership chain')).toBeInTheDocument();
  });

  it('applies low-confidence warning styling', () => {
    const lowConfidenceProps = {
      ...baseProps,
      confidence: 'Low' as const,
      confidenceScore: 35
    };
    render(<ProductResultScreen {...lowConfidenceProps} />);
    expect(screen.getByText(/Low \(35%\)/)).toBeInTheDocument();
  });

  it('renders confidence explanation and sources', () => {
    render(<ProductResultScreen {...baseProps} />);
    const tooltipButton = screen.getByText(/What affects this\?/i);
    expect(tooltipButton).toBeInTheDocument();
    expect(tooltipButton).toHaveAttribute('title', 'Confidence is based on data quality, source reliability, and agent agreement.');
  });

  it('matches snapshot', () => {
    const { container } = render(<ProductResultScreen {...baseProps} />);
    expect(container).toMatchSnapshot();
  });

  it('handles ultimate owner correctly', () => {
    const ultimateOwnerProps = {
      ...baseProps,
      ownershipTrail: [
        {
          name: 'Test Brand',
          country: 'United States',
          flag: '🇺🇸',
          type: 'Brand'
        },
        {
          name: 'Ultimate Owner',
          country: 'United States',
          flag: '🇺🇸',
          type: 'Parent Company',
          ultimate: true
        }
      ]
    };
    render(<ProductResultScreen {...ultimateOwnerProps} />);
    expect(screen.getAllByText('Ultimate Owner')).toHaveLength(2);
    expect(screen.getByText('🎯')).toBeInTheDocument();
  });

  it('renders different company types with correct styling', () => {
    const mixedTypesProps = {
      ...baseProps,
      ownershipTrail: [
        {
          name: 'Private Company',
          country: 'United States',
          flag: '🇺🇸',
          type: 'Private'
        },
        {
          name: 'Public Company',
          country: 'United States',
          flag: '🇺🇸',
          type: 'Public'
        },
        {
          name: 'State-owned Company',
          country: 'United States',
          flag: '🇺🇸',
          type: 'State-owned'
        }
      ]
    };
    render(<ProductResultScreen {...mixedTypesProps} />);
    expect(screen.getByText('Private')).toBeInTheDocument();
    expect(screen.getByText('Public')).toBeInTheDocument();
    expect(screen.getByText('State-owned')).toBeInTheDocument();
  });

  it('toggles trace visibility', () => {
    render(<ProductResultScreen {...baseProps} />);
    const showButton = screen.getByText(/Show Steps/i);
    
    // Initially trace should be hidden
    expect(screen.queryByText('Web Search')).not.toBeInTheDocument();
    
    // Click to show
    fireEvent.click(showButton);
    expect(screen.getByText('Web Search')).toBeInTheDocument();
    expect(screen.getByText('Hide Steps')).toBeInTheDocument();
    
    // Click to hide
    fireEvent.click(screen.getByText('Hide Steps'));
    expect(screen.queryByText('Web Search')).not.toBeInTheDocument();
  });

  it('toggles manual entry form', () => {
    render(<ProductResultScreen {...baseProps} />);
    const manualEntryButton = screen.getByText(/Enter Manually/i);
    
    // Initially form should be hidden
    expect(screen.queryByPlaceholderText('Product Name')).not.toBeInTheDocument();
    
    // Click to show
    fireEvent.click(manualEntryButton);
    expect(screen.getByPlaceholderText('Product Name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Brand Name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Company Name')).toBeInTheDocument();
  });

  it('shows confidence tooltip on hover', () => {
    render(<ProductResultScreen {...baseProps} />);
    const tooltipButton = screen.getByText(/What affects this\?/i);
    expect(tooltipButton).toHaveAttribute('title', 'Confidence is based on data quality, source reliability, and agent agreement.');
  });

  it('renders ownership trail arrows', () => {
    render(<ProductResultScreen {...baseProps} />);
    // Check for arrow separators between ownership steps
    const arrows = screen.getAllByText('→');
    expect(arrows.length).toBeGreaterThan(0);
  });

  it('handles error state', () => {
    const errorProps = {
      ...baseProps,
      error: 'Failed to fetch ownership data'
    };
    render(<ProductResultScreen {...errorProps} />);
    expect(screen.getByText('❌ Error')).toBeInTheDocument();
    expect(screen.getByText('Failed to fetch ownership data')).toBeInTheDocument();
    expect(screen.getAllByText('🔄 Retry')).toHaveLength(2);
  });

  it('handles fallback state', () => {
    const fallbackProps = {
      ...baseProps,
      fallback: 'Using cached data from previous search'
    };
    render(<ProductResultScreen {...fallbackProps} />);
    expect(screen.getByText('🧯 Fallback')).toBeInTheDocument();
    expect(screen.getByText('Using cached data from previous search')).toBeInTheDocument();
  });
});

describe('ProductResultScreen trace and verification', () => {
  // Create simplified test props based on trace mocks
  const createTestProps = (result: any) => ({
    product: {
      name: result.product_name || 'Test Product',
      brand: result.brand || 'Test Brand',
      country: 'United States',
      flag: '🇺🇸',
      barcode: result.barcode
    },
    ownershipTrail: result.ownership_chain?.map((chain: any) => ({
      name: chain.name,
      country: chain.country,
      flag: '🇺🇸',
      type: chain.role
    })) || [],
    confidence: (result.confidence_score >= 75 ? 'High' : result.confidence_score >= 50 ? 'Medium' : 'Low') as 'High' | 'Medium' | 'Low',
    confidenceScore: result.confidence_score || 0,
    sources: result.sources || [],
    processTrace: result.agent_execution_trace?.stages?.map((stage: any) => ({
      step: stage.stage,
      status: stage.status,
      reasoning: `Stage ${stage.stage} completed`,
      time: `${stage.duration_ms}ms`
    })) || [],
    error: null,
    fallback: null
  });

  it('shows correct trace for manual entry', () => {
    const props = createTestProps(mockTraceManualEntry);
    const { getByText, queryByText } = render(<ProductResultScreen {...props} />);
    
    // Show trace steps
    const showButton = getByText(/Show Steps/i);
    fireEvent.click(showButton);
    
    expect(getByText('manual_entry')).toBeInTheDocument();
    expect(queryByText('vision')).not.toBeInTheDocument();
  });

  it('shows retry in trace when retry happened', () => {
    const props = createTestProps(mockTraceRetry);
    const { getByText } = render(<ProductResultScreen {...props} />);
    
    // Show trace steps
    const showButton = getByText(/Show Steps/i);
    fireEvent.click(showButton);
    
    // Should show web_research steps (appears twice in retry scenario)
    expect(screen.getAllByText(/web_research/i)).toHaveLength(4);
  });

  it('shows verified pill when verification_status is verified', () => {
    const props = createTestProps(mockTraceVerified);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(getByText(/88%/)).toBeInTheDocument();
    expect(getByText(/High/)).toBeInTheDocument();
  });

  it('shows warning pill when verification_status needs verification', () => {
    const props = createTestProps(mockTraceUnverified);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(getByText(/42%/)).toBeInTheDocument();
    expect(getByText(/Low/)).toBeInTheDocument();
  });

  it('renders alternatives[] when present', () => {
    const props = createTestProps(mockTraceAmbiguous);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(getByText('OK Snacks A/S')).toBeInTheDocument();
    expect(mockTraceAmbiguous.alternatives.length).toBeGreaterThan(0);
  });

  it('renders unknown fallback correctly', () => {
    const props = createTestProps(mockTraceUnknown);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(getByText(/Low \(0%\)/)).toBeInTheDocument();
  });

  it('renders confidence pill with correct color', () => {
    const props = createTestProps(mockTraceVerified);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(getByText(/88%/)).toBeInTheDocument();
  });

  it('renders sources in correct section', () => {
    const props = createTestProps(mockTraceVerified);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(screen.getByText(/What affects this\?/)).toBeInTheDocument();
  });

  it('generates headline/story only after final results', () => {
    const spy = vi.spyOn(global.console, 'log');
    const props = createTestProps(mockTraceVerified);
    render(<ProductResultScreen {...props} />);
    expect(spy).not.toHaveBeenCalledWith(expect.stringContaining('headline fired multiple times'));
    spy.mockRestore();
  });

  it('matches snapshot for verified result', () => {
    const props = createTestProps(mockTraceVerified);
    const { container } = render(<ProductResultScreen {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('handles empty ownership chain gracefully', () => {
    const emptyResult = {
      ...mockTraceUnknown,
      ownership_chain: []
    };
    const props = createTestProps(emptyResult);
    const { getByText } = render(<ProductResultScreen {...props} />);
    expect(getByText('🧬 Ownership')).toBeInTheDocument();
  });

  it('displays trace steps with proper status indicators', () => {
    const props = createTestProps(mockTraceVerified);
    const { getByText } = render(<ProductResultScreen {...props} />);
    const showButton = getByText(/Show Steps/i);
    fireEvent.click(showButton);
    
    // Check that trace steps are displayed
    expect(screen.getAllByText(/manual_entry/i)).toHaveLength(2);
    expect(screen.getAllByText(/web_research/i)).toHaveLength(2);
    expect(screen.getAllByText(/ownership_analysis/i)).toHaveLength(2);
  });

  it('shows performance metrics in trace', () => {
    const props = createTestProps(mockTraceVerified);
    const { getByText } = render(<ProductResultScreen {...props} />);
    const showButton = getByText(/Show Steps/i);
    fireEvent.click(showButton);
    
    // Check for time information in trace
    expect(screen.getAllByText(/ms/i)).toHaveLength(6);
  });
}); 