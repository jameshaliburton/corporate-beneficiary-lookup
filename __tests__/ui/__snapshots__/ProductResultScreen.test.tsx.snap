// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductResultScreen > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col min-h-screen bg-gray-50 pb-24"
  >
    <section
      class="bg-white rounded-xl shadow p-6 mb-4 mt-4 mx-2"
    >
      <div
        class="flex flex-col items-center text-center"
      >
        <span
          class="text-3xl font-bold mb-1"
        >
          Test Product
        </span>
        <span
          class="text-lg text-gray-600 mb-1"
        >
          Test Brand
        </span>
        <span
          class="text-base flex items-center gap-1 mb-1"
        >
          🇺🇸
           
          United States
        </span>
        <span
          class="text-xs text-gray-400"
        >
          Barcode: 
          123456789
        </span>
      </div>
    </section>
    <section
      class="bg-white rounded-xl shadow p-4 mb-4 mx-2"
    >
      <div
        class="mb-2 font-semibold text-lg flex items-center gap-2"
      >
        🧬 Ownership
      </div>
      <div
        class="flex flex-wrap gap-2 justify-center"
      >
        <div
          class="flex items-center gap-1 px-3 py-2 rounded-full shadow-sm border bg-gray-100 text-gray-800"
          style="font-weight: normal; border-width: 1px;"
        >
          <span>
            Test Brand
          </span>
          <span>
            🇺🇸
          </span>
          <span
            class="ml-1 px-2 py-0.5 rounded text-xs font-medium bg-white border border-gray-200"
          >
            Brand
          </span>
          <span
            class="mx-2 text-gray-400"
          >
            →
          </span>
        </div>
        <div
          class="flex items-center gap-1 px-3 py-2 rounded-full shadow-sm border bg-gray-100 text-gray-800"
          style="font-weight: normal; border-width: 1px;"
        >
          <span>
            Parent Company Inc
          </span>
          <span>
            🇺🇸
          </span>
          <span
            class="ml-1 px-2 py-0.5 rounded text-xs font-medium bg-white border border-gray-200"
          >
            Parent Company
          </span>
        </div>
      </div>
    </section>
    <section
      class="bg-white rounded-xl shadow p-4 mb-4 mx-2"
    >
      <div
        class="mb-2 font-semibold text-lg flex items-center gap-2"
      >
        📊 Confidence
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span
          class="px-3 py-1 rounded-full font-semibold text-sm bg-blue-100 text-blue-800"
        >
          High
           (
          92
          %)
        </span>
        <button
          class="ml-2 text-xs underline text-blue-600"
          title="Confidence is based on data quality, source reliability, and agent agreement."
        >
          What affects this?
        </button>
      </div>
      <div
        class="mt-2 flex gap-2"
      >
        <button
          class="px-3 py-1 rounded bg-green-100 text-green-800 font-medium text-xs"
        >
          🙋 Confirmed
        </button>
        <button
          class="px-3 py-1 rounded bg-red-100 text-red-800 font-medium text-xs"
        >
          ❗ Flag Incorrect
        </button>
      </div>
    </section>
    <section
      class="bg-white rounded-xl shadow p-4 mb-4 mx-2"
    >
      <div
        class="flex items-center justify-between mb-2"
      >
        <span
          class="font-semibold text-lg flex items-center gap-2"
        >
          🧪 Trace
        </span>
        <button
          class="text-xs underline text-blue-600"
        >
          Show
           Steps
        </button>
      </div>
    </section>
    <div
      class="fixed bottom-0 left-0 w-full bg-white border-t shadow-lg flex justify-around py-3 z-50"
    >
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-blue-100 text-blue-800 font-medium"
      >
        🔄 Retry
      </button>
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-green-100 text-green-800 font-medium"
      >
        🙋 Confirm
      </button>
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-red-100 text-red-800 font-medium"
      >
        ❗ Flag
      </button>
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-gray-100 text-gray-800 font-medium"
      >
        📝 Enter Manually
      </button>
    </div>
  </div>
</div>
`;

exports[`ProductResultScreen trace and verification > matches snapshot for verified result 1`] = `
<div>
  <div
    class="flex flex-col min-h-screen bg-gray-50 pb-24"
  >
    <section
      class="bg-white rounded-xl shadow p-6 mb-4 mt-4 mx-2"
    >
      <div
        class="flex flex-col items-center text-center"
      >
        <span
          class="text-3xl font-bold mb-1"
        >
          Test Product
        </span>
        <span
          class="text-lg text-gray-600 mb-1"
        >
          Test Brand
        </span>
        <span
          class="text-base flex items-center gap-1 mb-1"
        >
          🇺🇸
           
          United States
        </span>
      </div>
    </section>
    <section
      class="bg-white rounded-xl shadow p-4 mb-4 mx-2"
    >
      <div
        class="mb-2 font-semibold text-lg flex items-center gap-2"
      >
        🧬 Ownership
      </div>
      <div
        class="flex flex-wrap gap-2 justify-center"
      >
        <div
          class="flex items-center gap-1 px-3 py-2 rounded-full shadow-sm border bg-gray-100 text-gray-800"
          style="font-weight: normal; border-width: 1px;"
        >
          <span>
            OK Snacks A/S
          </span>
          <span>
            🇺🇸
          </span>
          <span
            class="ml-1 px-2 py-0.5 rounded text-xs font-medium bg-white border border-gray-200"
          >
            Brand
          </span>
          <span
            class="mx-2 text-gray-400"
          >
            →
          </span>
        </div>
        <div
          class="flex items-center gap-1 px-3 py-2 rounded-full shadow-sm border bg-gray-100 text-gray-800"
          style="font-weight: normal; border-width: 1px;"
        >
          <span>
            Kims A/S
          </span>
          <span>
            🇺🇸
          </span>
          <span
            class="ml-1 px-2 py-0.5 rounded text-xs font-medium bg-white border border-gray-200"
          >
            Parent
          </span>
          <span
            class="mx-2 text-gray-400"
          >
            →
          </span>
        </div>
        <div
          class="flex items-center gap-1 px-3 py-2 rounded-full shadow-sm border bg-gray-100 text-gray-800"
          style="font-weight: normal; border-width: 1px;"
        >
          <span>
            Orkla ASA
          </span>
          <span>
            🇺🇸
          </span>
          <span
            class="ml-1 px-2 py-0.5 rounded text-xs font-medium bg-white border border-gray-200"
          >
            Ultimate Owner
          </span>
        </div>
      </div>
    </section>
    <section
      class="bg-white rounded-xl shadow p-4 mb-4 mx-2"
    >
      <div
        class="mb-2 font-semibold text-lg flex items-center gap-2"
      >
        📊 Confidence
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span
          class="px-3 py-1 rounded-full font-semibold text-sm bg-blue-100 text-blue-800"
        >
          High
           (
          88
          %)
        </span>
        <button
          class="ml-2 text-xs underline text-blue-600"
          title="Confidence is based on data quality, source reliability, and agent agreement."
        >
          What affects this?
        </button>
      </div>
      <div
        class="mt-2 flex gap-2"
      >
        <button
          class="px-3 py-1 rounded bg-green-100 text-green-800 font-medium text-xs"
        >
          🙋 Confirmed
        </button>
        <button
          class="px-3 py-1 rounded bg-red-100 text-red-800 font-medium text-xs"
        >
          ❗ Flag Incorrect
        </button>
      </div>
    </section>
    <section
      class="bg-white rounded-xl shadow p-4 mb-4 mx-2"
    >
      <div
        class="flex items-center justify-between mb-2"
      >
        <span
          class="font-semibold text-lg flex items-center gap-2"
        >
          🧪 Trace
        </span>
        <button
          class="text-xs underline text-blue-600"
        >
          Show
           Steps
        </button>
      </div>
    </section>
    <div
      class="fixed bottom-0 left-0 w-full bg-white border-t shadow-lg flex justify-around py-3 z-50"
    >
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-blue-100 text-blue-800 font-medium"
      >
        🔄 Retry
      </button>
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-green-100 text-green-800 font-medium"
      >
        🙋 Confirm
      </button>
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-red-100 text-red-800 font-medium"
      >
        ❗ Flag
      </button>
      <button
        class="flex-1 mx-2 px-3 py-2 rounded bg-gray-100 text-gray-800 font-medium"
      >
        📝 Enter Manually
      </button>
    </div>
  </div>
</div>
`;
